src/app/screens/signBatchPsbtRequest/index.tsx

```diff
@@ -0,0 +1,763 @@
+  paddingTop: props.theme.spacing(12),
+  paddingBottom: props.theme.spacing(20),
+}));
+
+const HeadingText = styled.h1((props) => ({
+  ...props.theme.headline_s,
+  color: props.theme.colors.white_0,
+  textAlign: 'center',
+  marginTop: props.theme.spacing(8),
+}));
+
+const BodyText = styled.h1((props) => ({
+  ...props.theme.body_m,
+  color: props.theme.colors.white_400,
+  marginTop: props.theme.spacing(8),
+  textAlign: 'center',
+  overflowWrap: 'break-word',
+  wordWrap: 'break-word',
+  wordBreak: 'break-word',
+  marginLeft: props.theme.spacing(5),
+  marginRight: props.theme.spacing(5),
+}));
+
+const CloseButton = styled(ActionButton)((props) => ({
+  marginTop: props.theme.spacing(58),
+}));
+
+const CircularProgressContainer = styled.div((props) => ({
+  padding: props.theme.spacing(3),
+}));
+
+function SignBatchPsbtRequest() {
+  const { btcAddress, ordinalsAddress, selectedAccount, network, btcFiatRate } =
+    useWalletSelector();
+  const navigate = useNavigate();
+  const { t } = useTranslation('translation', { keyPrefix: 'CONFIRM_TRANSACTION' });
+  // const { t: signatureRequestTranslate } = useTranslation('translation', {
+  //   keyPrefix: 'SIGNATURE_REQUEST',
+  // });
+  const [expandInputOutputView, setExpandInputOutputView] = useState(false);
+  const {
+    payload,
+    confirmSignPsbt,
+    broadcastPsbt,
+    cancelSignPsbt,
+    getSigningAddresses,
+    requestToken,
+  } = useSignBatchPsbtTx();
+  const [isSigning, setIsSigning] = useState(false);
+  const [isBroadcasting, setIsBroadcasting] = useState(false);
+  const [isBroadcastingComplete, setIsBroadcastingComplete] = useState(false);
+  const [isSigningComplete, setIsSigningComplete] = useState(false);
+  const [signingPsbtIndex, setSigningPsbtIndex] = useState(0);
+  const [broadcastingPsbtIndex, setBroadcastingPsbtIndex] = useState(0);
+  const [hasOutputScript, setHasOutputScript] = useState(false);
+  const [currentPsbtIndex, setCurrentPsbtIndex] = useState(0);
+  const [reviewTransaction, setReviewTransaction] = useState(false);
+  // const [isModalVisible, setIsModalVisible] = useState(false);
+  // const [currentStepIndex, setCurrentStepIndex] = useState(0);
+  // const [isButtonDisabled, setIsButtonDisabled] = useState(false);
+  // const [isConnectSuccess, setIsConnectSuccess] = useState(false);
+  // const [isConnectFailed, setIsConnectFailed] = useState(false);
+  // const [isTxApproved, setIsTxApproved] = useState(false);
+  // const [isTxRejected, setIsTxRejected] = useState(false);
+  const { search } = useLocation();
+  const params = new URLSearchParams(search);
+  const tabId = params.get('tabId') ?? '0';
+  // const request = decodeToken(requestToken) as any as SignTransactionOptions;
+  // const btcClient = useBtcClient();
+
+  const handlePsbtParsing = useCallback(
+    (item) => {
+      try {
+        return parsePsbt(selectedAccount!, item.inputsToSign, item.psbtBase64, network.type);
+      } catch (err) {
+        return '';
```

Comment:
instead of returning an empty string here we should probably handle this error instantly by displaying the error message screen with the invalid psbt index

Reply:
Makes sense, here's what it'll look like:
![Monosnap Xverse Wallet 2023-11-09 20-15-50](https://github.com/secretkeylabs/xverse-web-extension/assets/36603049/a493de0b-b31a-41f7-a101-00e794df0427)